arcgis-ios-sdk-samples/Content Display Logic/Controllers/ContentTableViewController.swift

```diff
@@ -28,29 +28,51 @@ class ContentTableViewController: UITableViewController {
     var allSamples = [Sample](){
         didSet{
             displayedSamples = allSamples
-            searchEngine = SampleSearchEngine(samples: allSamples)
+            if !containsSearchResults{
+                searchEngine = SampleSearchEngine(samples: allSamples)
+            }
         }
     }
-    private var expandedRowIndex:Int = -1
-    
-    var containsSearchResults = false
-    private var bundleResourceRequest:NSBundleResourceRequest!
-    private var downloadProgressView:DownloadProgressView!
+    private var expandedRowIndex: Int = -1
     
-    var searchEngine: SampleSearchEngine?
-    
-    // strong reference needed for iOS 10
-    var filterSearchController:UISearchController?
+    private var bundleResourceRequest: NSBundleResourceRequest!
+    private var downloadProgressView: DownloadProgressView!
     
     override func viewDidLoad() {
         super.viewDidLoad()
         
         //initialize download progress view
         downloadProgressView = DownloadProgressView()
         downloadProgressView.delegate = self
+    }
+    
+    // MARK: - Search
+    
+    var containsSearchResults = false
+    private var searchEngine: SampleSearchEngine?
+    
+    // strong reference needed for iOS 10
+    private var filterSearchController: UISearchController?
+    
+    private var hasSearchController: Bool{
+        if #available(iOS 11.0, *) {
+            return navigationItem.searchController != nil
+        }
+        else {
+            return navigationItem.titleView is UISearchBar
+        }
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
         
-        if !containsSearchResults{
-            addFilterSearchController()
+        if !containsSearchResults,
+            !hasSearchController {
+            // wait a little to avoid a crash
+            DispatchQueue.main.asyncAfter(deadline: .now()+0.1) { [weak self] () in
```

Comment:
Is there an event that could be used here instead of just delaying?

Reply:
Not that I could find. Using `viewDidLoad` or `viewWillAppear` both cause the visual glitch. However, it turns out the waiting is unnecessary, though the async is still needed to avoid a crash.